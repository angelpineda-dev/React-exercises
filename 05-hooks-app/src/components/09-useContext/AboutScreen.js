import { useContext } from "react";
import { UserContext } from "./UserContext";

const AboutScreen = () => {
  const { setUser } = useContext(UserContext);

  const handleClick = () => {
    setUser({});
  };
  return (
    <div>
      <h2>About</h2>
      <button className="btn btn-outline-primary" onClick={handleClick}>
        Logout
      </button>
    </div>
  );
};

export default AboutScreen;
