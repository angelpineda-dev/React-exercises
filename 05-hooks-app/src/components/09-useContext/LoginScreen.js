import { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const userData = {
    id: 123,
    name: "Angel Pineda",
  };

  return (
    <div>
      <h2>Login</h2>
      <button
        className="btn btn-outline-primary"
        onClick={() => setUser(userData)}
      >
        Login
      </button>
    </div>
  );
};
