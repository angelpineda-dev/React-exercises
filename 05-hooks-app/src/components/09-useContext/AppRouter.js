import { HashRouter, Switch, Route, Link } from "react-router-dom";
import AboutScreen from "./AboutScreen";
import HomeScreen from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import NavBar from "./NavBar";
const AppRouter = () => {
  return (
    <HashRouter>
      <div>
        <NavBar />
        <Switch>
          <Route exact path="/" component={HomeScreen} />
          <Route exact path="/about" component={AboutScreen} />
          <Route exact path="/login" component={LoginScreen} />
          <Route
            path="*"
            children={
              <div>
                <h1>Error 404 - Not Found</h1>
                <Link to="/">Regresar</Link>
              </div>
            }
          ></Route>
        </Switch>
      </div>
    </HashRouter>
  );
};

export default AppRouter;
