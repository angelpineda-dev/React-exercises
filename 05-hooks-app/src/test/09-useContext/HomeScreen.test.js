import HomeScreen from "../../components/09-useContext/HomeScreen";
import { UserContext } from "../../components/09-useContext/UserContext";
import { mount } from "enzyme";

describe("Pruebas en <HomeScreen/>", () => {
  const user = {
    name: "Angel",
    email: "email@domain.com",
  };

  test("Debe de mostrar el componente correctamente", () => {
    const wrapper = mount(
      <UserContext.Provider value={{ user }}>
        <HomeScreen />
      </UserContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
