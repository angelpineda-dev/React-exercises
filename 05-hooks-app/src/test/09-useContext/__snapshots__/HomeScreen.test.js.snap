// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen/> Debe de mostrar el componente correctamente 1`] = `
<div>
  <h2>
    Home
  </h2>
  <pre>
    {
   "name": "Angel",
   "email": "email@domain.com"
}
  </pre>
</div>
`;
